server/boards/app/files.go

```diff
@@ -175,3 +190,150 @@ func (a *App) MoveFile(channelID, teamID, boardID, filename string) error {
+	}
+
+	var destBoard *model.Board
+
+	for _, block := range copiedBlocks {
+		if block.Type != model.TypeImage && block.Type != model.TypeAttachment {
+			continue
+		}
+
+		fileId, isOk := block.Fields["fileId"].(string)
+		if !isOk {
+			fileId, isOk = block.Fields["attachmentId"].(string)
+			if !isOk {
+				continue
+			}
+		}
+
+		// create unique filename
+		ext := filepath.Ext(fileId)
+		fileInfoID := utils.NewID(utils.IDTypeNone)
+		destFilename := fileInfoID + ext
+
+		if destBoard == nil || block.BoardID != destBoard.ID {
+			destBoard = sourceBoard
+			if block.BoardID != destBoard.ID {
+				destBoard, err = a.GetBoard(block.BoardID)
+				if err != nil {
+					return fmt.Errorf("cannot fetch destination board %s for CopyCardFiles: %w", sourceBoardID, err)
+				}
+			}
+		}
+
+		// GetFilePath will retrieve the correct path
+		// depending on whether FileInfo table is used for the file.
+		fileInfo, sourceFilePath, err := a.GetFilePath(sourceBoard.TeamID, sourceBoard.ID, fileId)
+		if err != nil {
+			return fmt.Errorf("cannot fetch destination board %s for CopyCardFiles: %w", sourceBoardID, err)
+		}
+
+		destinationFilePath := filepath.Join(utils.GetBaseFilePath(), destFilename)
+		// Global Templates are handled via Import, if user-defined templates
+		// are to be stored by team. Won't be deleted by Data Retention
+		if asTemplate {
+			destinationFilePath = filepath.Join(destBoard.TeamID, destBoard.ID, destFilename)
+		}
+		if fileInfo == nil {
+			ext = filepath.Ext(sourceFilePath)
+			now := utils.GetMillis()
+			fileInfo = &mm_model.FileInfo{
+				Id:              fileInfoID[1:],
+				CreatorId:       "boards",
+				PostId:          emptyString,
+				ChannelId:       emptyString,
+				CreateAt:        now,
+				UpdateAt:        now,
+				DeleteAt:        0,
+				Path:            destinationFilePath,
+				ThumbnailPath:   emptyString,
+				PreviewPath:     emptyString,
+				Name:            destFilename,
+				Extension:       ext,
+				Size:            0,
+				MimeType:        emptyString,
+				Width:           0,
+				Height:          0,
+				HasPreviewImage: false,
+				MiniPreview:     nil,
+				Content:         "",
+				RemoteId:        nil,
+			}
+		} else {
+			fileInfo.Id = fileInfoID[1:]
+			fileInfo.Path = destinationFilePath
+		}
+		err = a.store.SaveFileInfo(fileInfo)
+		if err != nil {
+			return fmt.Errorf("CopyCardFiles: cannot create fileinfo: %w", err)
+		}
+
+		a.logger.Debug(
+			"Copying card file",
+			mlog.String("sourceFilePath", sourceFilePath),
+			mlog.String("destinationFilePath", destinationFilePath),
+		)
+
+		if err := a.filesBackend.CopyFile(sourceFilePath, destinationFilePath); err != nil {
+			a.logger.Error(
+				"CopyCardFiles failed to copy file",
+				mlog.String("sourceFilePath", sourceFilePath),
+				mlog.String("destinationFilePath", destinationFilePath),
+				mlog.Err(err),
+			)
+		}
+		block.Fields["fileId"] = destFilename
```

Comment:
Usually I try to avoid mutating my arguments in go. I don't usually expect it.

Reply:
Rewrote, returning new filenames rather than updating blocks...